### PR TITLE
feat(grid): Add <Row> and <Column> components

### LINKED ISSUE #1091: Website Grid has to be correct
I'm not sure if this should be solved by working on website grid, or website shell. But we need to make sure things are on grid for faster future work. We can't keep faking it.

### LINKED ISSUE #1269: bug[do/don't component grid alignment]: Misalignment at Max breakpoint
## bug[do/don't component grid alignment]: Misalignment at Max breakpoint

- [ ] Fix alignment at max breakpoint

<br>

### Correct:

At Xlg breakpoints and below, the Do / Don't component is aligning with the large image size

<br>

### Incorrect:

At Max breakpoint, the Do / Don't component is NOT aligning with the large image size

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #1091, closes #1269

Replaces `<GridWrapper>` with `<Row>` and `<Column>` components to allow more flexibility moving forward. 

#### Changelog

**New**

`<Row>`  `<Row className="custom-class">`

`<Column>`

`<Column>` props | description
-- | --
colSm | Specify the col width at small breakpoint
colMd | Specify the col width at medium breakpoint
colLg | Specify the col width at large breakpoint
offsetSm | Specify the col offset at small breakpoint
offsetMd | Specify the col offset at medium breakpoint
offsetLg | Specify the col offset at large breakpoint, defaults to 4
noGutterSm | Specify a no-gutter class at small breakpoint
noGutterMd | Specify a no-gutter class at medium breakpoint
noGutterLg | Specify a no-gutter class at large breakpoint
noGutterSmLeft | Specify a no-gutter left class at small breakpoint
noGutterMdLeft | Specify a no-gutter left class at medium breakpoint
noGutterLgLeft | Specify a no-gutter left class at large breakpoint
className | Set a custom class

**Changed**

- [x] Update all mdx files to use new grid components
- [x] Update docs
- Update <DoDontExample> styles since it sits inside column now at full width. Adds a new square prop for the text only versions on the content guidance page. 

**Removed**

- [x] remove `<GridWrapper>` component
- PageIcon component that wasn't being used 

### Example
For example this

```
<GridWrapper col_lg="8" flex="true" bleed="true">
  <ClickableTile .... />
  <ClickableTile .... />
</GridWrapper>
```

would now be written as
```
<Row className="tile--group">
<Column lg="4" bleedSm>
  <ClickableTile .... />
</Column>
<Column lg="4" offsetLg="0" bleedSm>
  <ClickableTile .... />
</Column>
</Row>
```
@carbon-design-system/design & @carbon-design-system/ui would love feedback on this before I go and convert the site over to these new grid components.

### CODE REVIEW

**Comment by joshblack** [comment]
Makes sense to me 👍 Especially when used alongside MDX. One thing around bleedLeft/right, would it handle auto-swapping the bleed in RTL? That was one nice thing about the hang terminology is that if we swapped language direction then it knew which gutter to drop.

File: src/components/Grid/Grid.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      bleedMd,
+      bleedLg,
+      bleedLeftSm,
+      bleedLeftMd,
+      bleedLeftLg,
+      className,
+    } = this.props;
+
+    let colClasses = '';
+    if (lg) colClasses += `ibm--col-lg-${lg} `;
```
Comment (asudoh) [inline src/components/Grid/Grid.js]: Weighing in - I think we can use `classnames` package here.

Comment (alisonjoseph) [reply src/components/Grid/Grid.js]: I ran into an issue trying to use it, I'll take another look and ping you. thanks for reviewing!

**Comment by alisonjoseph** [comment]
> One thing around bleedLeft/right, would it handle auto-swapping the bleed in RTL? That was one nice thing about the hang terminology is that if we swapped language direction then it knew which gutter to drop.

Not currently 😉 but you bring up a good point. I think looking at RTL support for the grid, in general, is a larger issue. The offset
screen shot 2019-02-26 at 9 03 20 am
 classes need to be reversed also I think.

File: src/components/Grid/Grid.js (hunk 1 above)
Comment (alisonjoseph) [reply src/components/Grid/Grid.js]: Updated @asudoh 👍

File: packages/addons-website/package.json [hunk 2]
```diff
@@ -23,15 +23,19 @@
   },
   "devDependencies": {
     "@babel/core": "^7.2.2",
+    "babel-loader": "7.1.1",
     "@babel/preset-env": "^7.2.0",
     "@babel/preset-react": "^7.0.0",
+    "@babel/plugin-syntax-dynamic-import": "^7.2.0",
```
Comment (asudoh) [inline packages/addons-website/package.json]: Do we actually use dynamic import (`import()` as a function) syntax?

File: packages/addons-website/scss/components/_website-grid.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  .container .ibm--row .ibm--row {
+    margin: 0;
+  }
+  
+  .container .ibm--row .ibm--row > div {
+    margin: 0;
+    max-width: 100%;
+    flex: auto;
+  }
+
```
Comment (asudoh) [inline packages/addons-website/scss/components/_website-grid.scss]: Probably missing end curly bracket?

File: packages/addons-website/package.json (hunk 2 above)
Comment (alisonjoseph) [reply packages/addons-website/package.json]: removed this 👍

File: packages/addons-website/scss/components/_website-grid.scss (hunk 3 above)
Comment (alisonjoseph) [reply packages/addons-website/scss/components/_website-grid.scss]: fixed

**Comment by alisonjoseph** [comment]
Yes @asudoh we should probably add testing, I'm just not sure where to start. (I'm out tomorrow-next Monday but can look at that more next week sometime possibly)

**Comment by asudoh** [comment]
Cool please don't hesitate to let me know if you want to add tests in a separate PR. (Not sure how quickly team wants to move forward with this PR, just double-checking)

**Comment by lovemecomputer** [comment]
this is def a big improvement to putting elements on a page! would love to catch up tomorrow about the convo points and review then :)

**Comment by alisonjoseph** [comment]
Thanks @asudoh I think with the push for v10 right now, might be best to push test to a separate PR.

File: docs/CONTENT.md [hunk 4]
```diff
@@ -148,6 +148,32 @@ The carbon website has a handful of custom markdown components available for use
 
 ### Global Components:
 
+### Row & Column
+`<Column>` props | description
+-- | --
+sm | Specify the col width at small breakpoint
+md | Specify the col width at medium breakpoint
+lg | Specify the col width at large breakpoint
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply docs/CONTENT.md]: Oh good catch! I missed the doc update

**Comment by shixiedesign** [comment]
Hey everyone. I tried my best to review with limited understanding of how things should be affected. Here are the visual differences I noticed between preview link and live site. They may or may not be due to this fixes in this PR so please strike those out of they were not meant to be fixed here: 

- [ ] Do-don't examples on Motion page have issues. 1. Should all be squares with video taking up all space inside. 2. caption padding is too much here. Might not be a fix in this PR. **3. All Don't examples have turned into Do!** 1 and 3 are important issues to fix. 
image
- [ ] These guys look like they got left out of the grid structure. They should be 12 col wide. 
image
- [ ] Code snippets in preview site (top) seems to have smaller type size than live site (bottom)
image
- [ ] Tab cuts into the tab content area
image
- [ ] UI shell image gone again 😂 I know y'all have fixed this 100 times.
image

**Comment by alisonjoseph** [comment]
Most fixes made except

Code snippet - I updated to latest from components, so possibly a change there? this would be separate PR issue either way if something looks off. 

Tabs - should they not hang left? I didn't touch anything with tabs in this PR so just checking.

**Comment by jnm2377** [comment]
Re: Codesnippet, font-size hasn't changed in Carbon Components. It's `12px`. Inspected the live site and the preview here and they're both `12px`... 🤷‍♀️

**Comment by jnm2377** [comment]
Also re: tabs, I think she's talking about a small overlap in the corner. I missed it at first.
tabs

**Comment by jnm2377** [comment]
Though, I agree that tabs could be a separate issue since this PR is for something else.

**Comment by shixiedesign** [comment]
- [ ] Ooo this distance from left nav definitely looks too big! Also everything looks zoomed in. I double checked it's not my browser. Resource tiles also look bigger. No image here. This could be the reason code snippet above looked bigger. Top is live, bottom is preview link:
image

**Comment by shixiedesign** [comment]
I suddenly occured to me that the code snippet type style looks bigger because IT IS NOT PLEX MONO! This should be addressed before Mike notices 😂 Should i make a new issue?

**Comment by alisonjoseph** [comment]
@joshblack can you take a look at this when you have time (obviously release stuff comes first). Trying to add the grid stuff to the add on repo and thats what is making it pull in 12 col grid for some reason. I'm stumped.

**Comment by vpicone** [comment]
This is great! I implemented something similar for the theme but just with the most common row style, really helpful for the default markdown components.

File: packages/addons-website/src/components/Grid/Grid.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+   */
+  offsetMd: PropTypes.string,
+  /**
+   * Specify the col offset at small breakpoint
+   */
+  offsetSm: PropTypes.string,
+  /**
+   * Specify a no-gutter class
+   */
+  noGutterSm: PropTypes.boolean,
```
Comment (vpicone) [inline packages/addons-website/src/components/Grid/Grid.js]: PropTypes.bool

File: packages/addons-website/src/components/Grid/Grid.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+   */
+  colMd: PropTypes.string,
+  /**
+   * Specify the col width at large breakpoint
+   */
+  colLg: PropTypes.string,
+  /**
+   * Specify the col offset at large breakpoint
+   */
+  offsetLg: PropTypes.string,
```
Comment (vpicone) [inline packages/addons-website/src/components/Grid/Grid.js]: Can we make these numbers?

File: packages/addons-website/src/components/Grid/Grid.js (hunk 5 above)
Comment (alisonjoseph) [reply packages/addons-website/src/components/Grid/Grid.js]: oops, good catch

**Comment by alisonjoseph** [comment]
updated @vpicone

**Comment by alisonjoseph** [comment]
@shixiedesign wanted to see if you had time to re-review this before merging.

**Pull request merged**